### PR TITLE
wr-130: Fix facebook integration

### PR DESCRIPTION
**Pull request by IrynaK-a**
Replace react-facebook-login (that not supports react@18) with react-facebook

### CODE REVIEW

File: frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx [hunk 1]
```diff
@@ -17,39 +11,59 @@ type FacebookLoginButtonProperties = {
...
+      }
+
+      onLogin({
+        accessToken: response.authResponse.accessToken,
+        email: profile.email,
+      });
+    } catch (error) {
+      error instanceof Error && notification.error(error.message);
+    }
+  }, [onLogin, init]);
```
Comment (drMangust) [inline frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx]: What if we put it in a separate hook?
But this is just an observation, there are no specific reasons for this, it just looks like this code can be moved from here.

File: frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -9,10 +12,7 @@ import {
 
 import { AuthSubmitButton } from '../auth-submit-button/auth-submit-button.js';
 import { AuthSignInButton } from '../components.js';
-/**
- * @todo should be fixed with react-facebook package
- */
-// import { FacebookLoginButton } from '../facebook-login-button/facebook-login-button.js';
+import { FacebookLoginButton } from '../facebook-login-button/facebook-login-button.js';
 import { PasswordInput } from '../password-input/password-input.js';
```
Comment (drMangust) [inline frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx]:
```suggestion
import { 
  AuthSubmitButton, 
  AuthSubmitButton, 
  FacebookLoginButton, 
  PasswordInput,
} from '../components.js';
```

File: frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 3]
```diff
@@ -49,7 +49,9 @@ const SignInForm: React.FC<Properties> = ({
           onClick={onGoogleLogin}
           label="Sign in with Google"
         ></AuthSignInButton>
-        {/* <FacebookLoginButton onLogin={onFacebookLogin} /> */}
+        <FacebookProvider appId={config.ENV.FACEBOOK.APP_ID}>
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx]: Can we wrap with FacebookProviderr FacebookLoginButton in it's component?

File: frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx (hunk 1 above)
Comment (andanf-e) [reply frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx]: That's a specific method needed for specific button, so I'm not sure that we need to move it out

File: frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx [hunk 4]
```diff
@@ -17,39 +11,59 @@ type FacebookLoginButtonProperties = {
...
+      if (!profile?.email) {
+        throw new Error(FACEBOOK_ERROR_MESSAGE);
+      }
+
+      onLogin({
+        accessToken: response.authResponse.accessToken,
+        email: profile.email,
+      });
+    } catch (error) {
+      error instanceof Error && notification.error(error.message);
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/facebook-login-button/facebook-login-button.tsx]:
```suggestion
      if (error instanceof Error) {
        notification.error(error.message);
      }
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: frontend/src/pages/auth/components/facebook-login-button/facebook-login.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { FacebookProvider } from 'react-facebook';
+
+import { config } from '~/libs/packages/config/config.js';
+import { type UserSignInWithFacebookResponseDto } from '~/packages/auth/auth.js';
+
+import { FacebookLoginButton } from '../components.js';
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/facebook-login-button/facebook-login.tsx]: But it looks like this button component is placed in same folder, are you sure you need to import it from `components` folder

**Review by andanf-e — DISMISSED** [review]

**Review by mhrechkin — DISMISSED** [review]

**Review by andanf-e — APPROVED** [review]

**Pull request merged**